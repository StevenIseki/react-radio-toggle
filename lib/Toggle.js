'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Toggle = function (_React$Component) {
  _inherits(Toggle, _React$Component);

  function Toggle() {
    _classCallCheck(this, Toggle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Toggle).apply(this, arguments));
  }

  _createClass(Toggle, [{
    key: 'clear',
    value: function clear() {
      var elem = _react2.default.findDOMNode(this);
      var checkedElem = $(elem).find('input:checked')[0];
      if (checkedElem !== undefined) {
        checkedElem.checked = false;
      }
    }
  }, {
    key: 'handleClick',
    value: function handleClick(e) {
      var parentLabel = e.target.parentElement;
      if (parentLabel !== undefined) {
        var value = parentLabel.querySelector('span').innerText;
        if (this.props.onToggle !== undefined) {
          this.props.onToggle(value);
        }
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props;
      var items = _props.items;
      var checked = _props.checked;
      var name = _props.name;


      var radioItems = items.map(function (item, i) {
        if (item === checked) {
          return _react2.default.createElement(
            'label',
            { key: i },
            _react2.default.createElement('input', { type: 'radio', defaultChecked: true, name: name,
              onClick: _this2.handleClick.bind(_this2) }),
            _react2.default.createElement(
              'span',
              null,
              item
            )
          );
        } else {
          return _react2.default.createElement(
            'label',
            { key: i },
            _react2.default.createElement('input', { type: 'radio', name: name,
              onClick: _this2.handleClick.bind(_this2) }),
            _react2.default.createElement(
              'span',
              null,
              item
            )
          );
        }
      });

      return _react2.default.createElement(
        'div',
        { id: name, className: 'toggle' },
        radioItems
      );
    }
  }]);

  return Toggle;
}(_react2.default.Component);

exports.default = Toggle;


Toggle.propTypes = {
  onToggle: _react2.default.PropTypes.func
};