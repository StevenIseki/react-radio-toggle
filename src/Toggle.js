import React, { Component, PropTypes } from 'react'
import { render } from 'react-dom'

export default class Toggle extends React.Component {

  clear () {
    let elem = React.findDOMNode(this)
    let checkedElem = $(elem).find('input:checked')[0]
    if(checkedElem !== undefined) {
      checkedElem.checked = false
    }
  }

  handleClick(e) {
    let parentLabel = e.target.parentElement
    if(parentLabel !== undefined){
      let value = parentLabel.querySelector('span').innerText
      if (this.props.onToggle !== undefined) {
        this.props.onToggle(value)
      }
    }
  }

  render() {
    const { items, checked, name } = this.props;

    let radioItems = items.map( (item, i) => {
      if(item === checked){
        return (
          <label key={i}>
            <input type="radio" defaultChecked name={name}
                   onClick={this.handleClick.bind(this)} />
            <span>{item}</span>
          </label>
        )
      } else {
        return (
          <label key={i}>
            <input type="radio" name={name}
                   onClick={this.handleClick.bind(this)} />
            <span>{item}</span>
          </label>
        )
      }
    })

    return (
      <div id={name} className="toggle">
        {radioItems}
      </div>
    );
  }
}

Toggle.propTypes = {
  onToggle: React.PropTypes.func
}
