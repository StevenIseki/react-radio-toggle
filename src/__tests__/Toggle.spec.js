/* setup enzyme */
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

/* setup jsdom */
var jsdom = require('jsdom')
const { JSDOM } = jsdom
const window = new JSDOM('').window
global.window = window
global.document = window.document

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Toggle from '../Toggle'

test('Toggle renders correctly and matches snapshot', () => {
  let items = ['apple', 'orange', 'lemon', 'peach']
  const handleToggled = jest.fn()

  const component = renderer.create(
    <Toggle
      items={items}
      checked='apple'
      name='toggleFruits'
      onToggle={handleToggled}
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Toggle renders the correct elements and props', () => {
  let items = ['apple', 'orange', 'lemon', 'peach']
  const handleToggled = jest.fn()

  const wrapper = shallow(
    <Toggle
      items={items}
      checked='apple'
      name='toggleFruits'
      onToggle={e => handleToggled(e)}
    />
  )

  expect(wrapper.instance().props.items).toEqual(items)
  expect(wrapper.instance().props.checked).toEqual('apple')
  expect(wrapper.instance().props.name).toEqual('toggleFruits')

  expect(wrapper.find('label').length).toEqual(items.length)
  expect(wrapper.find('input').length).toEqual(items.length)
  expect(wrapper.find('input').first().prop('defaultChecked')).toEqual(true)
  expect(wrapper.find('input').first().prop('type')).toEqual('radio')
  expect(wrapper.find('span').first().text()).toEqual('apple')

  wrapper
    .find('input')
    .first()
    .simulate('click', { target: {} })

  // expect(handleToggled).toBeCalled();
})
