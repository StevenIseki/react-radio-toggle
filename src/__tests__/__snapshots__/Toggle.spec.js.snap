// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly and matches snapshot 1`] = `
<div
  className="sc-bdVaJa kVuGjP"
  id="toggleFruits"
>
  <label>
    <input
      defaultChecked={true}
      name="toggleFruits"
      onClick={[Function]}
      type="radio"
    />
    <span>
      apple
    </span>
  </label>
  <label>
    <input
      name="toggleFruits"
      onClick={[Function]}
      type="radio"
    />
    <span>
      orange
    </span>
  </label>
  <label>
    <input
      name="toggleFruits"
      onClick={[Function]}
      type="radio"
    />
    <span>
      lemon
    </span>
  </label>
  <label>
    <input
      name="toggleFruits"
      onClick={[Function]}
      type="radio"
    />
    <span>
      peach
    </span>
  </label>
</div>
`;
